sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py

```diff
@@ -67,6 +68,9 @@ async def upload(
             await super()._upload(rule_id, stream=stream_name, body=logs, content_encoding=content_encoding, **kwargs)
             return
 
+        if not isinstance(logs, Sequence):
+            raise ValueError("The 'logs' parameter must be a list of JSON objects or an I/O stream that is readable.")
```

Comment:
What was the old behavior if logs is not Sequence?

Reply:
Broken behavior. If a dict was passed in, the dict will be iterated through with what pretty much amounts to just gzipping the keys of the dict and losing the data.